### DISCUSSION TITLE
How are prebuilts clang and sysroot built for host system?

### DISCUSSION THREAD

**Discussion by Biswa96** [android/ndk#2047] [Q&A] [answered]
Please let me know if there is any documentation about the aforementioned question.

**Comment by pirama-arumuga-nainar** [comment, marked as answer]
https://android.googlesource.com/toolchain/llvm_android/+/refs/heads/main/BUILD.md has instructions to bulid clang.  The llvm-toolchain repo includes sysroots for the host (for windows and linux).

**Reply by Biswa96** [reply]
I am using exactly that repository. But it depends on prebuilts clang directory as mentioned here https://android.googlesource.com/toolchain/llvm_android/+/refs/heads/llvm-r522817/paths.py#42. I am asking about that clang toolchain.

**Reply by Biswa96** [reply]
I have figured out the following steps. Please correct me if I am wrong.
* have system-toolchain
* build stage1-llvm using system-toolchain
* build stage2-llvm using stage1-llvm

But I can not figure out how the sysroot was created at first place. I can not find any repository for the headers specifically.

**Reply by pirama-arumuga-nainar** [reply]
If you do `repo init ...` followed by `reop sync` like mentioned in `BUILD.md`, it should create the prebuilts/clang directory (and anything else necessary to run build.py).
